onboarding-enabler-spring/src/main/java/org/zowe/apiml/enable/config/EnableApiDiscoveryConfig.java

```diff
@@ -35,8 +39,28 @@ public MessageService messageServiceDiscovery() {
     }
 
     @Bean
-    public ApiMediationClient apiMediationClient() {
-       return new ApiMediationClientImpl();
+    public ApiMediationClient apiMediationClient(@Nullable EurekaClientProvider eurekaClientProvider) {
```

Comment:
How did you get to this change? 
 - What does it support?

Reply:
This method is obsolete and I'll remove it. It is superseded by the one with more parameters.

The idea is to be able to create a single bean of type ApiMediationClientImpl which can be constructed with variable number of parameters which are themselves optional beans. If a Bean is not provided then default one is constructed i the enabler. If a EurekaClientProvider bean exists it will be used by the enabler. Similarly for EurekaClientConfigProvider.

Reply:
I understand what it does. 

I don't understand what it supports, like what is the let's say business need that led you to this solution.

Reply:
Let's just get it in and if we later find out some problems with it we will figure out how to refactor.